File: pm_core_1/src/infrastructure/store/model/issue_type_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type IssueTypeModel struct {
	BaseModel
	TenantID       int64   `gorm:"column:tenant_id;not null" json:"tenantId"`
	TypeKey        string  `gorm:"column:type_key;type:varchar(20);not null" json:"typeKey"`
	Name           string  `gorm:"column:name;type:varchar(100);not null" json:"name"`
	Description    *string `gorm:"column:description" json:"description,omitempty"`
	IconUrl        *string `gorm:"column:icon_url" json:"iconUrl,omitempty"`
	HierarchyLevel int     `gorm:"column:hierarchy_level;not null" json:"hierarchyLevel"`
	IsSystem       bool    `gorm:"column:is_system;not null" json:"isSystem"`
}

func (IssueTypeModel) TableName() string {
	return "issue_types"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/IssueTypeModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "issue_types")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class IssueTypeModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "type_key", nullable = false)
    private String typeKey;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "icon_url")
    private String iconUrl;

    @Column(name = "hierarchy_level")
    private Integer hierarchyLevel;

    @Column(name = "is_system")
    private Boolean isSystem;
}
